developer-docs/architecture/import.md

```diff
@@ -0,0 +1,144 @@
+# Importing Resources
+
+There are a variety of scenarios that require the ability for users to import existing
+resources for management by Pulumi. For example:
+
+- Migrating from manually-managed resources to IaC
+- Migrating from other IaC platforms to Pulumi
+- Migrating resources between Pulumi stacks
+
+At a minimum, importing a resource involves adding the resource's state to the destination
+stack's statefile. Once the resource has been added to the stack, the Pulumi CLI is able
+to manage the resource like any other. In order to do anything besides delete the resource,
+however, the user must also add a definition for the resource to their Pulumi program.
+
+Both of the import approaches used by Pulumi aim to prevent the accidental modification or
+deletion of a resource being imported. Though the user experiences of these approaches are
+quite different, they share a common principle: at the point at which a resource is
+successfully imported, the stack's Pulumi program must contain a definition for the
+resource that accurately describes its current state (i.e. there are no differences
+between the state described in the program and the actual state of the imported resource).
+
+## `import` resource option
+
+The oldest method supported of importing resources into a stack is the `import` resource
+option. When set, this option specifies the ID of an existing resource to import into the
+stack. The exact behavior of this option depends on the current state of the resource
+within the destination stack:
+
+1. If the resource does not exist, it is imported
+2. If the resource exists and has the same `ID` or `ImportID`, the resource is treated
+   like any other resource
+3. Otherwise, the current resource is deleted and replaced by importing the resource with
+   the specified ID
+
+The trickiest of these three situations is (2). This state transition is intended to allow
+users to import a resource and then continue to make changes to their program without
+requiring that they remove the resource option. For example, this allows a user to import
+a resource in one `pulumi up`, then successfully run another `pulumi up` without removing
+the `import` option from their program and without attempting to import the resource a
+second time.
+
+As mentioned in [the introduction](#importing-resources), the `import` resource option
+requires that the desired state described by Pulumi program for a resource being imported
+matches the actual state of the resource as returned by the provider. More precisely,
+given a resource R of type T with  import ID X and the resource inputs present in the
```

Comment:
```suggestion
given a resource `R` of type `T` with import ID `X` and the resource inputs present in the
```

Reply:
Ah, ok. It's hard to read in the markdown, but whatever renders nicely in the docs is fine.